climpyrical/tests/test_gridding.py

```diff
@@ -50,6 +57,32 @@ def test_check_ndims(data, n, error):
 xext_bad, yext_bad = np.repeat(xi, yi.size), np.tile(yi, xi.size)
 
 
+@pytest.mark.parametrize(
+    "ds,dv,n,keys,error",
+    [
+        (ds, dv, 3, {"p"}, KeyError),
+        (3, dv, 3, {"rlat", "rlon", "lon", "lat", "level"}, TypeError),
+        (ds, 3, 3, {"rlat", "rlon", "lon", "lat", "level"}, TypeError),
+        (ds, dv, "4", {"rlat", "rlon", "lon", "lat", "level"}, TypeError),
+        (ds, dv, 3, {"rlat", "rlon", "lon", "lat", "level"}, None),
+    ],
+)
+def test_check_regrid_ensemble_inputs(ds, dv, n, keys, error):
+    if error is None:
+        check_regrid_ensemble_inputs(ds, dv, n, keys)
+    else:
+        with pytest.raises(error):
+            check_regrid_ensemble_inputs(ds, dv, n, keys)
+
+
+@pytest.mark.parametrize(
+    "ds,dv,n,keys", [(ds, dv, 3, {"rlat", "rlon", "lat", "lon", "level"})]
+)
```

Comment:
Do you plan to have other cases? I see a mix here of parametrization on args to `regrid_ensemble`, but comparing to a fixed result (`ds_regridded_proper`). That is not often done. Usually it's all one way (params) or the other (fixed values).